subprojects/performance/src/templates/generateLotsOfDeprecationWarnings/build.gradle

```diff
@@ -20,9 +21,9 @@ void createDeprecations(int iterations) {
     for (int i=0; i<iterations; i++) {
         // every 10th deprecation should have a different message
         if(i%10 == 0) {
-            org.gradle.util.DeprecationLogger.nagUserOfDeprecated("Some unique deprecation No #\$i")
+            DeprecationLogger.deprecate("Some unique deprecation No #\$i").undocumented().nagUser()
         }
-        org.gradle.util.DeprecationLogger.nagUserOfDeprecated("Some repetitive Deprecation")
+        DeprecationLogger.deprecate("Some repetitive Deprecation").undocumented().nagUser()
```

Comment:
I don't believe this will work as-is because these performance tests can be run against older versions of Gradle that won't have these APIs.  It looks like this test is against 4.9 and up.  All of the old versions of Gradle are run less often (once a week) and very late in the pipeline.

CompileStatic makes this pretty difficult to do right and making this code dynamic might change what we're testing.

I would try removing CompileStatic and have two versions of createDeprecations.  One that uses the old API and one that uses the new API and then below check if gradle.gradleVersion >= 6.2 and use the new API.  We'd want to see if making this dynamic greatly impacts what we're measuring. If it does, we should just keep the old API around a bit longer until we can raise the minimum tested version to 6.2.  It probably doesn't make much sense to compare ancient versions of Gradle for this test.

Reply:
Let's leave it for a few more versions then until we can raise the minimum tested-against to 6.2